app/actions/app_apply_manifest.rb

```diff
@@ -188,21 +193,15 @@ def create_binding_message(service_instance_guid, app_guid, manifest_service_bin
       )
     end
 
-    def raise_binding_error!(service_instance, message)
-      error_message = "For service '#{service_instance.name}': #{message}"
-      raise ServiceBindingError.new(error_message)
-    end
-
-    def binding_exists?(service_instance, app)
-      binding = ServiceBinding.first(service_instance: service_instance, app: app)
-      binding && !binding.create_failed?
-    end
-
-    def binding_being_deleted!(service_instance, app)
-      binding = ServiceBinding.first(service_instance: service_instance, app: app)
-      if binding && binding.operation_in_progress? && binding.last_operation.type == 'delete'
-        raise_binding_error!(service_instance, 'An existing binding is being deleted. Try recreating the binding later.')
```

Comment:
I changed the error message to fit all other error messages better and to not need a huge switch case with custom error messages. However this means the message changes, dont know but maybe this should not be changed as its part of the API and people(cf cli) check this string explicitly ? Opinions ?

Reply:
Also on "create failed" bindings we recreate that binding. Wouldnt it make sense to actually also do this for the "delete failed" case since orphan mitigation anyway tries to delete that binding and one can assume (since the apply_manifest wants it) a new binding is requested ? So we could retrigger a binding on failed operation and throw an error just on "in progress" ones as one has to wait a bit for them to bind completely.